### PR TITLE
Feature - cart order ui

### PR DESCRIPTION
**Pull request by at-lytruong**

### CODE REVIEW

**Comment by at-daiho** [comment]
:+1: There are no linter violations.

File: MyApp/View/Controllers/Cart/CartViewController.swift [hunk 1]
```diff
@@ -10,7 +10,52 @@ import UIKit
 
 class CartViewController: UIViewController {
 
+    @IBOutlet weak var tableView: UITableView!
+    @IBOutlet weak var priceTotalLabel: UILabel!
     override func viewDidLoad() {
         super.viewDidLoad()
+
+        self.title = "Cart"
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Cart/CartViewController.swift]: Remove self

File: MyApp/View/Controllers/Cart/CartViewController.swift [hunk 2]
```diff
@@ -10,7 +10,52 @@ import UIKit
...
+        tableView.delegate = self
+    }
+
+    @IBAction func orderTouchUpInSide(_ sender: Any) {
+
+        let orderVC = OrderViewController()
+        orderVC.modalTransitionStyle = .crossDissolve
+        orderVC.modalPresentationStyle = .custom
+        orderVC.view.isOpaque = false
+        self.present(orderVC, animated: true, completion: nil)
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Cart/CartViewController.swift]: self.

File: MyApp/View/Controllers/Cart/CartViewController.swift [hunk 3]
```diff
@@ -10,7 +10,52 @@ import UIKit
...
+    }
+}
+// MARK: - TableViewDataSoure
+extension CartViewController: UITableViewDataSource {
+
+    func numberOfSections(in tableView: UITableView) -> Int {
+
+        return 1
+    }
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Cart/CartViewController.swift]: Remove func numberOfSections

File: MyApp/View/Controllers/Cart/CartViewController.swift [hunk 4]
```diff
@@ -10,7 +10,52 @@ import UIKit
...
+        tableView.dataSource = self
+        tableView.delegate = self
+    }
+
+    @IBAction func orderTouchUpInSide(_ sender: Any) {
+
+        let orderVC = OrderViewController()
+        orderVC.modalTransitionStyle = .crossDissolve
+        orderVC.modalPresentationStyle = .custom
+        orderVC.view.isOpaque = false
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Cart/CartViewController.swift]: Don't set "isOpaque = false" in here

File: MyApp/View/Controllers/Cart/OrderViewController.swift [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        orderView.backgroundColor = UIColor.black.withAlphaComponent(0.6)
+        alertOrderView.insertSubview(imageView, aboveSubview: childView)
+    }
+
+    @IBAction func checkOutOrderTouchUpInSide(_ sender: Any) {
+    }
+
+    @IBAction func cancelTouchUpInSide(_ sender: Any) {
+
+        self.dismiss(animated: true, completion: nil)
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Cart/OrderViewController.swift]: self

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 6]
```diff
@@ -10,7 +10,113 @@ import UIKit
...
+    func initTableView() {
+        tableView.register(UINib(nibName: "HomeTableViewCell", bundle: nil), forCellReuseIdentifier: "CellTableView")
+
+        tableView.register(UINib(nibName: "TrenddingHomeTableViewCell", bundle: nil), forCellReuseIdentifier: "CellTrendding")
+
+        tableView.delegate = self
+        tableView.dataSource = self
+    }
+    func initCollectionView() {
+        collectionView.register(UINib(nibName: "HomeCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "CellCollection")
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: Using in SwiftUtils

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 7]
```diff
@@ -10,7 +10,113 @@ import UIKit
...
+        let rightNaviBar = UIBarButtonItem(customView: button)
+        self.navigationItem.rightBarButtonItem = rightNaviBar
+
+        let leftNaviBar = UIBarButtonItem(title: App.String.categories, style: .plain, target: self, action: #selector(HomeViewController.categories))
+        self.navigationItem.leftBarButtonItem = leftNaviBar
+    }
+    func initTableView() {
+        tableView.register(UINib(nibName: "HomeTableViewCell", bundle: nil), forCellReuseIdentifier: "CellTableView")
+
+        tableView.register(UINib(nibName: "TrenddingHomeTableViewCell", bundle: nil), forCellReuseIdentifier: "CellTrendding")
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: SwiftUtils

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 8]
```diff
@@ -10,7 +10,113 @@ import UIKit
...
+        button.addTarget(self, action: #selector(HomeViewController.searchBarItem), for: UIControlEvents.touchUpInside)
+        button.frame = CGRect(x: 0, y: 0, width: 30, height: 30)
+        let rightNaviBar = UIBarButtonItem(customView: button)
+        self.navigationItem.rightBarButtonItem = rightNaviBar
+
+        let leftNaviBar = UIBarButtonItem(title: App.String.categories, style: .plain, target: self, action: #selector(HomeViewController.categories))
+        self.navigationItem.leftBarButtonItem = leftNaviBar
+    }
+    func initTableView() {
+        tableView.register(UINib(nibName: "HomeTableViewCell", bundle: nil), forCellReuseIdentifier: "CellTableView")
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: SwiftUtils

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 9]
```diff
@@ -10,7 +10,113 @@ import UIKit
...
+}
+// MARK: - UICollectionViewDataSource
+extension HomeViewController: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+
+        return 3
+    }
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "CellCollection", for: indexPath) as? HomeCollectionViewCell else { return UICollectionViewCell() }
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: SwiftUtils

File: MyApp/View/Controllers/Home/HomeViewController.swift [hunk 10]
```diff
@@ -10,7 +10,113 @@ import UIKit
...
+    func numberOfSections(in tableView: UITableView) -> Int {
+        return 2
+    }
+
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return 1
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "CellTableView", for: indexPath) as? HomeTableViewCell else {
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Home/HomeViewController.swift]: SwiftUtils

File: MyApp/View/Controllers/Home/TableViewCell/CollectionViewCell/ListCollectionViewCell.swift [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+class ListCollectionViewCell: UICollectionViewCell {
+
+    @IBOutlet weak var itemImageView: UIImageView!
+    @IBOutlet weak var itemLabel: UILabel!
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        layer.borderWidth = 1
+        layer.cornerRadius = 8
+        layer.borderColor = UIColor.clear.cgColor
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Home/TableViewCell/CollectionViewCell/ListCollectionViewCell.swift]: SwiftUtils

File: MyApp/View/Controllers/Home/TableViewCell/HomeTableViewCell.swift [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+}
+
+class HomeTableViewCell: UITableViewCell {
+
+    @IBOutlet weak var collectionView: UICollectionView!
+
+    weak var delegate: HomeTableViewCellDelegate?
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        collectionView.register(UINib(nibName: "ListCollectionViewCell", bundle: nil), forCellWithReuseIdentifier: "cell")
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Home/TableViewCell/HomeTableViewCell.swift]: SwiftUtils

File: MyApp/View/Controllers/Home/TableViewCell/HomeTableViewCell.swift [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+}
+extension HomeTableViewCell: UICollectionViewDataSource {
+    func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return 1
+    }
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return 2
+    }
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: "cell", for: indexPath) as? ListCollectionViewCell else {
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Home/TableViewCell/HomeTableViewCell.swift]: SwiftUtils

File: MyApp/View/Controllers/Home/TableViewCell/HomeTableViewCell.swift [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+//  HomeTableViewCell.swift
+//  MyApp
+//
+//  Created by AST on 7/21/17.
+//  Copyright © 2017 Asian Tech Co., Ltd. All rights reserved.
+//
+
+import UIKit
+protocol HomeTableViewCellDelegate: class {
+    func HomeTableViewCellDelegate(_ collectionview: UICollectionView, didSelectItemAt indexPath: IndexPath)
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Home/TableViewCell/HomeTableViewCell.swift]: Naming

File: MyApp/View/Controllers/Cart/CartViewController.swift (hunk 4 above)
Comment (at-lytruong) [inline MyApp/View/Controllers/Cart/CartViewController.swift]: where is set it?

File: MyApp/View/Controllers/Cart/CartCell.swift [hunk 15]
```diff
@@ -0,0 +1,39 @@
...
+
+    var viewModel: CartCellViewModel? {
+        didSet {
+            updateView()
+        }
+    }
+    override func awakeFromNib() {
+        super.awakeFromNib()
+
+    }
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Cart/CartCell.swift]: remove unuse.

File: MyApp/View/Controllers/Home/CollectionViewCell/HomeCollectionViewCell.swift [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+import UIKit
+
+class HomeCollectionViewCell: UICollectionViewCell {
+
+    @IBOutlet weak var imageView: UIImageView!
+
+    override func awakeFromNib() {
+        super.awakeFromNib()
+        // Initialization code
+    }
```
Comment (at-hieutran) [inline MyApp/View/Controllers/Home/CollectionViewCell/HomeCollectionViewCell.swift]: remove unuse

**Comment by at-hieutran** [comment]
LGTM

**Pull request merged**